pkg/executor/push.go

```diff
@@ -115,6 +116,19 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 			continue
 		}
 
+		// Historically kaniko was pre-configured by default with gcr credential helper,
+		// in here we keep the backwards compatibility by enabling the GCR helper only
+		// when gcr.io is in one of the destinations.
+		if strings.Contains(destRef.RegistryStr(), "gcr.io") {
```

Comment:
Can you please test to push to a public GCR repo one needs to be authenticated? My gut is mostly yes in which case this is good. If not, then we might need to add this is documentation. 

It would be great if you can add tests for this function. 
1. You can declare 2 variables 
```
// for testing
var (
execute = execCommad
checkRemotePushPermissions = remote.CheckPushPermission
stat = os.Stat
)
```

In your tests, you could create mock for these and make sure, you 
1. When "/kaniko/.docker/config.json" exists, os.Stat is not called
2. When it does not exist os.Stat is called and mock for "execute" returns nil.

Reply:
I added the testing. I noticed we were already using [afero](https://github.com/spf13/afero) in push.go and push_test.go so there wasn't a need to mock os.Stat. Thank you so much for detailed recommendations on how to test, really helpful.

There is no way to make a GCR repo pushable by the public. You can only push if you're explicitly given permission to push to a GCR repo. For reference: https://cloud.google.com/container-registry/docs/access-control or did you mean something else?